subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/DistributionPluginIntegrationTest.groovy

```diff
@@ -130,7 +132,7 @@ class DistributionPluginIntegrationTest extends WellBehavedPluginTest {
             """
         then:
         runAndFail('customDistZip')
-        failure.assertHasCause "Distribution baseName must not be null or empty! Check your configuration of the distribution plugin."
+        failure.assertThatDescription(containsString("Distribution baseName must not be null or empty! Check your configuration of the distribution plugin."))
```

Comment:
I'm quite unhappy about these changes to exception hierarchy. What causes these?

Reply:
This actually reverts your changes to the exception hierarchy. The cause is that we do not evaluate the input file properties when building the task graph: https://github.com/gradle/gradle/pull/3787/commits/c6072a003c4666bba399eb0cf429ac549003595f
IOW, my changes would be 4.4 behavior, while current `master` exhibits a different behavior.

Reply:
We can revert my changes about the lazy evaluation again, now that we don't evaluate the outputs when building the task graph.
Then we have the same exception behavior as on `master`, but not the same as in `4.4`.
WDYT?

Reply:
With lazy evaluation:
```
* What went wrong:
Distribution baseName must not be null or empty! Check your configuration of the distribution plugin.
```
from `DefaultTaskExecutionPlan.getOutputPaths`

Without lazy evaluation:
```
* What went wrong:
Execution failed for task ':customDistZip'.
> Distribution baseName must not be null or empty! Check your configuration of the distribution plugin.
```
from [here](https://github.com/gradle/gradle/blob/5aa09a5b4964934aa02c8f51ed4c91a72f62a38b/subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskExecutionPlan.java#L611)

The message with the context (no lazy evaluation) looks better I think.